litellm/llms/snowflake/completion.py

```diff
@@ -0,0 +1,379 @@
+            obj = obj.strip()
+            if obj:
+                # Remove the 'data: ' prefix if it exists
+                if obj.startswith("data: "):
+                    obj = obj[6:]
+                # Load the JSON object into a Python dictionary
+                json_dict = json.loads(str(obj))
+                # Append the JSON dictionary to the list
+                json_list.append(json_dict)
+
+        completion = ""
+        model = ""
+        choices = {}
+        for chunk in json_list:
+
+            model = chunk["model"]
+            choices = chunk["choices"][0]
+
+            if "content" in choices["delta"].keys():
+
+                completion += choices["delta"]["content"]
+
+        processed_response = {
+            "model": model,
+            "completion": completion,
+            "stop_reason": choices["finish_reason"],
+        }
+
+        return ModelResponse(**processed_response)
+
+    async def async_completion(
+        self,
+        model: str,
+        model_response: ModelResponse,
+        api_base: str,
+        logging_obj,
+        encoding,
+        headers: dict,
+        data: dict,
+        client=None,
+    ):
+        if client is None:
+            client = AsyncHTTPHandler(timeout=httpx.Timeout(timeout=600.0, connect=5.0))
+
+        response = await client.post(api_base, headers=headers, data=json.dumps(data))
+
+        if response.status_code != 200:
+            raise SnowflakeError(
+                status_code=response.status_code, message=response.text
+            )
+
+        ## LOGGING
+        logging_obj.post_call(
+            input=data["content"],
+            api_key=headers.get("Authorization"),
+            original_response=response.text,
+            additional_args={"complete_input_dict": data},
+        )
+
+        response = self._process_response(
+            model_response=model_response,
+            response=response,
+            encoding=encoding,
+            prompt=data["content"],
+            model=model,
+        )
+        return response
+
+    async def async_streaming(
+        self,
+        model: str,
+        api_base: str,
+        logging_obj,
+        headers: dict,
+        data: Optional[dict],
+        client=None,
+    ):
+        if client is None:
+            client = AsyncHTTPHandler(timeout=httpx.Timeout(timeout=600.0, connect=5.0))
+
+        response = await client.post(api_base, headers=headers, data=json.dumps(data))
+
+        if response.status_code != 200:
+            raise SnowflakeError(
+                status_code=response.status_code, message=response.text
+            )
+
+        completion_stream = response.aiter_lines()
+
+        streamwrapper = CustomStreamWrapper(
+            completion_stream=completion_stream,
+            model=model,
+            custom_llm_provider="snowflake",
+            logging_obj=logging_obj,
+        )
+        return streamwrapper
+
+    def completion(
```

Comment:
if snowflake api is openai-compatible (can be called via openai sdk), does it need a custom calling config? 

why not just do it like this - https://github.com/BerriAI/litellm/blob/7630680690fdb85a0c239c49fa9817793b55447d/litellm/llms/azure_ai/chat/handler.py#L13